crates/extensions/tedge_mqtt_bridge/src/lib.rs

```diff
@@ -373,14 +443,21 @@ async fn half_bridge(
                 Incoming::PubAck(PubAck { pkid: ack_pkid })
                 | Incoming::PubRec(PubRec { pkid: ack_pkid }),
             ) => {
+                acknowledged += 1;
                 if let Some(msg) = forward_pkid_to_received_msg.remove(&ack_pkid) {
-                    let target = target.clone();
-                    tokio::spawn(async move { target.ack(&msg).await.unwrap() });
+                    if let Err(err) = target.ack(&msg).await {
```

Comment:
Can `ack` not block in the same way `publish` does?

Reply:
I previously observed blocked `ack` when the channel cycle was only made of bounded channels. Now that one of the channel of this cycle is unbounded, I don't think `ack` might block. However, to be sure, I'm missing a better understanding of `rumttc` internals.

That said, now that [BidirectionalChannelHalf owns the target](https://github.com/thin-edge/thin-edge.io/pull/3122/commits/5783f15acfbf034e51a15af51082c5aca25c8c9c), we can do the same for `ack` as for `pub`.
=> I will do that.